### DISCUSSION TITLE
TIMEOUT error when ./run_tests -j n

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#19922] [Q&A Getting Started] [answered]
Dear all,
  I have installed moose by the offline way on my workstation with Centos 7. When run ./run_tests -j n,  all the failed tests show [FINISHED] FAILED (TIMEOUT). I have tried to increase the timeout limit, but still failed.  When i just `./moose_test-opt`, the program hangs without any output.

`which mpi*` shows correct returns.

I found a similar issue #16788. The author solved the issue by commenting lines loading some modules for intel fortran that insludes openmpi in  `bash_profile`. But my `bash_profile` and `.bashrc` do not contain any lines relavant to openmpi. 

Any suggestions?
Best regards,

**Comment by GiudGiud** [comment, marked as answer]
Hello

Could you please run the program in gdb, interrupt it then get us a backtrace?
https://stackoverflow.com/questions/11616364/getting-a-stack-trace-of-a-hung-process-running-in-gdb

Guillaume

**Reply by bylore** [reply]
I ran `moose_test-opt` in gdb, after i interrupted it by Ctrl+C, i got the following message.

Starting program: /home/xxx/offline/moose/test/moose_test-opt moose_test-opt
warning: no loadable sections found in added symbol-file system-supplied DSO at 0x7ffff7ffd000
[Thread debugging using libthread_db enabled]
warning: File "/home/xxx/gcc/lib64/libstdc++.so.6.0.26-gdb.py" auto-loading has been declined by your `auto-load safe-path' set to "/usr/share/gdb/auto-load:/usr/lib/debug:/usr/bin/mono-gdb.py".
To enable execution of this file add
	add-auto-load-safe-path /home/xxx/gcc/lib64/libstdc++.so.6.0.26-gdb.py
line to your configuration file "/home/xxx/.gdbinit".
To completely disable this security protection add
	set auto-load safe-path /
line to your configuration file "/home/xxx/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
	info "(gdb)Auto-loading safe path"
^C

**Program received signal SIGINT, Interrupt.
0x0000003f9d80c465 in ?? () from /lib64/libpthread.so.0**

Missing separate debuginfos, use: debuginfo-install glibc-2.12-1.107.el6.x86_64 glibc-2.12-1.192.el6.x86_64 libX11-1.6.3-2.el6.x86_64 libXau-1.0.6-4.el6.x86_64 libudev-147-2.46.el6.x86_64 libudev-147-2.73.el6_8.2.x86_64 libxcb-1.11-2.el6.x86_64 libxml2-2.7.6-21.el6_8.1.x86_64 libxml2-2.7.6-8.el6_3.4.x86_64 zlib-1.2.3-29.el6.x86_64

**Reply by bylore** [reply]
using `backtrace` command, i got
`#0  0x0000003f9d80c465 in ?? () from /lib64/libpthread.so.`
`#1  0x00007ffff3aa8cd3 in libMesh::Singleton::Setup::Setup() () from /home/xxx/offline/moose/libmesh/installed/lib/libmesh_opt.so.0`
`#2  0x00007ffff3e2cdc6 in _GLOBAL__sub_I_remote_elem.C () from /home/xxx/offline/moose/libmesh/installed/lib/libmesh_opt.so.0`
`#3  0x00007ffff482c772 in __do_global_ctors_aux () from /home/xxx/offline/moose/libmesh/installed/lib/libmesh_opt.so.0`
`#4  0x00007fffffffca68 in ?? ()`
`#5  0x0000000000000002 in ?? ()`
`#6  0x00007fffffffca68 in ?? ()`
`#7  0x00007ffff392959b in _init () from /home/xxx/offline/moose/libmesh/installed/lib/libmesh_opt.so.0`
`#8  0x00007fffee4299c0 in ?? ()`
`#9  0x0000003f9cc0e705 in _dl_init_internal () from /lib64/ld-linux-x86-64.so.2`
`#10 0x0000003f9cc00b3a in _dl_start_user () from /lib64/ld-linux-x86-64.so.2`
`#11 0x0000000000000002 in ?? ()`
`#12 0x00007fffffffcda5 in ?? ()`
`#13 0x00007fffffffcde7 in ?? ()`
`#14 0x0000000000000000 in ?? ()`

**Reply by bylore** [reply]
To use moose, i added
`export LD_LIBRARY_PATH=/home/xxx/offline/moose/libmesh/installed/lib:$LD_LIBRARY_PATH` to my `.bashrc` file.
After doing this, my previous program built on external libmesh showed the similar hanging behavior.
And then i commented the above LD_LIBRARY_PATH line, my program built on externally installed libmesh worked well again.

So, the problem may be related to the libmesh installed asscociated with moose. 

I installed moose on another machine through the same way and the TIMEOUT error did not occur and most tests passed.

**Reply by GiudGiud** [reply]
What do you get by running `ldd ./executable-opt` ?
Do you get a different result when LD_LIBRARY_PATH is set to include libmesh/installed and when it is not?

**Reply by GiudGiud** [reply]
and just to make sure, you have only installed a single copy of libmesh on the entire system ?
